ReactSkia/sdk/WindowDelegator.cpp

```diff
@@ -61,16 +61,34 @@ void  WindowDelegator::createNativeWindow() {
 }
 
 void WindowDelegator::closeWindow() {
-  RNS_LOG_TODO("Sync between rendering & Exit to be handled ");
   windowActive = false;
-  std::scoped_lock lock(renderCtrlMutex);
 
   if(ownsTaskrunner_) {
+    windowTaskRunner_->dispatch([&](){
+      closeNativeWindow();
+      sem_post(&semReadyToDraw_);
```

Comment:
Can we have a separate semaphore for close Window Synchronization.
If close happen well before Expose Event occurs, then there would be 2 consumers for  "semReadyToDraw_" signals. That can possibly cause dead lock.

Reply:
sem_wait not required here . updated.